### PR TITLE
Implement new TDEM waveform (HalfSineWaveform and TriangularWaveform) & add waveform unit tests

### PR DESCRIPTION
**Pull request by ikding**
Resolves: #893 

This PR adds the following functionalities for SimPEG: 

* Two new SimPEG waveforms, `HalfSineWaveform` and `TriangularWaveform`. 
  * `HalfSineWaveform` uses a quarter sine ramp-on and ramp-off (instead of linear ramp off, like the case for `QuarterSineRampOnWaveform`). 
  * `TriangularWaveform` was originally left by SimPEG as an "exercise to the reader"; I made it inherit `TrapezoidWaveform` so we can just reuse the same code.
* Unit tests for most TDEM waveforms (which I couldn't find in the SimPEG repo, so adding them for completeness)

I also updated the TDEM waveform example code. Here is a schematic that includes the two new waveforms:

Screen Shot 2020-06-15 at 8 09 49 PM

Produced by: https://github.com/KoBoldMetals/simpeg/blob/5b23589492ed178bad684644c32fe5906ce9a217/examples/06-tdem/plot_fwd_tdem_waveforms.py

### CODE REVIEW

File: SimPEG/electromagnetics/time_domain/sources.py [hunk 1]
```diff
@@ -153,6 +143,21 @@ def eval(self, time):
             return 0
 
 
+class TriangularWaveform(TrapezoidWaveform):
+    """
+    TriangularWaveform is a special case of TrapezoidWaveform where there's no pleateau
```
Comment (lheagy) [inline SimPEG/electromagnetics/time_domain/sources.py:148]: :+1: Thanks for this addition!

File: tests/em/tdem/test_TDEM_sources.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    RampOffWaveform,
+    VTEMWaveform,
+    TrapezoidWaveform,
+    TriangularWaveform,
+    QuarterSineRampOnWaveform,
+    HalfSineWaveform,
+)
+
+
+class TestStepOffWaveform(unittest.TestCase):
```
Comment (lheagy) [inline tests/em/tdem/test_TDEM_sources.py:16]: this is excellent @ikding! Thanks for raising the bar 😄

**Comment by ikding** [comment]
Thanks @lheagy ! I'll let your team decide when to merge this. This PR fails the Travis-CI check due to errors in generating galleries (https://travis-ci.org/github/simpeg/simpeg/jobs/705557505), but I was not sure how to fix it (or whether the error is due to my PR vs. some other unrelated issue.)

**Comment by lheagy** [comment]
looks like the docs testing is timing out... I can rebuild locally and update the cache we have (cc @jcapriot)

**Comment by jcapriot** [comment]
I cannot for the life of me figure out why this is stalling on the doc build. I've uploaded the corresponding tar files from this branch, it all builds locally and it completes just fine...

**Comment by lheagy** [comment]
hmmm.... it looks like it is still trying to generate the gallery.

**Comment by lheagy** [comment]
@jcapriot: is this the 3.6 / 3.7 issue with the docs again? did you build on 3.6 or 3.7?

**Comment by jcapriot** [comment]
Nope, I built it on my 3.6 environment

**Comment by lheagy** [comment]
hmm... should we just remove doc testing for now until we get this figured out? And you or I can test for now locally before merging? it is unfortunate that this keeps popping up as a blocker for prs...

**Comment by jcapriot** [comment]
Possibly, I’ll start looking into azure-pipelines a little closer too

**Comment by ikding** [comment]
Oops, I accidentally pushed more commits to this PR because I merged another PR into `KoBoldMetals/simpeg`. Did not realize the side effect. I'll try to revert it.

**Pull request merged**